LazySignal.compute will now pass kwargs to dask.array.Array.compute

magnunor opened the pull request:
Currently, `kwargs` are not passed from `s.compute` to the dask `compute` function which is done on the data.

There are several useful parameters like `scheduler='processes'` and `num_workers=4`, which is useful when working with large datasets.

So this pull request simply passes `kwargs` in `s.compute` to `s.data.compute()`. It also improves the docstring for `compute`, and adds a test.

### Progress of the PR
- [x] Change implemented
- [x] update docstring
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add test
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import dask.array as da
data = da.zeros((100, 100, 100), chunks=(10, 20, 20))
s = hs.signals.Signal2D(data).as_lazy()
s.compute(scheduler="processes", num_workers=2)
```

ericpre commented on hyperspy/_signals/lazy.py:
```diff
@@ -241,19 +241,38 @@ def compute(self, close_file=False, show_progressbar=None, **kwargs):
+        Using processes scheduler and 2 workers
+
+        >>> s2 = s.deepcopy()
+        >>> s2.compute(scheduler="processes", num_workers=2)
```
While you are at it, it is worth adding a link to https://docs.dask.org/en/stable/scheduling.html#configuration in a `notes` or `see also` section to inform users of alternative way to set compute settings?

magnunor replied:
Done! Added it to `Notes`

jlaehne commented on hyperspy/_signals/lazy.py:
```diff
@@ -241,19 +241,38 @@ def compute(self, close_file=False, show_progressbar=None, **kwargs):
         %s
+        **kwargs : dict
+            Any other keyword arguments for :py:func:`dask.array.Array.compute`.
+            For example scheduler or num_workers.
```
```suggestion
            For example `scheduler` or `num_workers`.
```

magnunor replied:
Done!

magnunor commented:
Sorry about the slow response! I rebased this onto the newest `RELEASE_next_minor`, and fixed the docstring improvements suggest by @jlaehne

Or should this rather be sent to `RELEASE_next_major`?

ericpre commented on hyperspy/_signals/lazy.py:
```diff
@@ -236,23 +236,42 @@ def compute(self, close_file=False, show_progressbar=None, **kwargs):
+
         """
         if "progressbar" in kwargs:
             warnings.warn(
                 "The `progressbar` keyword is deprecated and will be removed "
                 "in HyperSpy 2.0. Use `show_progressbar` instead.",
                 VisibleDeprecationWarning,
             )
-            show_progressbar = kwargs["progressbar"]
+            show_progressbar = kwargs.pop("progressbar")
```
In the `RELEASE_next_major`, it can be removed.

ericpre reviewed (commented):
Yes, I think it will more simple if it goes directly to `RELEASE_next_major`.

magnunor committed:
LazySignal.compute will now pass kwargs to dask.array.Array.compute

So that parameters like scheduler and num_workers can be set when
using s.compute()

magnunor commented:
All done!
